src/disco/containers/InstallButton.js

```diff
@@ -29,12 +32,54 @@ export class InstallButton extends React.Component {
     downloadProgress: 0,
   }
 
+  componentDidMount() {
+    const { dispatch, guid, installURL, slug } = this.props;
+    this.addonManager = new AddonManager(guid, installURL, this.statusChanged);
+    this.addonManager.getAddon()
+      .then(() => {
+        dispatch({
+          type: 'INSTALL_STATE',
+          payload: {slug, guid, url: installURL, status: INSTALLED}});
+      })
+      .catch(() => {
```

Comment:
Always resolving in `getAddon()` sounds good to me. I was expecting it to yield an add-on object that had a state on it when I imagined it. Looks like `getAddonById()` just returns something falsey in that case, which is cool too.